source/dem/particle_particle_linear_force.cc

```diff
@@ -393,40 +383,73 @@ PPLinearForce<dim>::calculate_linear_contact_force_and_torque(
   // Check for gross sliding
   if (tangential_force.norm() > coulomb_threshold)
     {
-      // Gross sliding occurs and the tangential overlap and tangnetial
+      // Gross sliding occurs and the tangential overlap and tangential
       // force are limited to Coulumb's criterion
       contact_info.tangential_overlap =
         (coulomb_threshold *
            (tangential_force / (tangential_force.norm() + DBL_MIN)) -
-         dashpot_tangential_force) /
+         damping_tangential_force) /
         (tangential_spring_constant + DBL_MIN);
 
       tangential_force =
         (tangential_spring_constant * contact_info.tangential_overlap) +
-        dashpot_tangential_force;
+        damping_tangential_force;
     }
 
   // Calculation of torque
   // Torque caused by tangential force (tangential_torque)
   if (dim == 3)
     {
-      tangential_torque =
-        cross_product_3d((0.5 *
-                          particle_one_properties[DEM::PropertiesIndex::dp] *
-                          normal_unit_vector),
-                         tangential_force);
+      particle_one_tangential_torque =
+        cross_product_3d(normal_unit_vector,
+                         tangential_force * particle_one_properties[DEM::dp] *
+                           0.5);
+      particle_two_tangential_torque =
+        particle_one_tangential_torque *
+        particle_two_properties[DEM::PropertiesIndex::dp] /
+        particle_one_properties[DEM::PropertiesIndex::dp];
     }
 
   // Rolling resistance torque
-  rolling_resistance_torque = (this->*calculate_rolling_resistance_torque)(
-    this->effective_radius,
-    particle_one_properties,
-    particle_two_properties,
-    this->effective_coefficient_of_rolling_friction[particle_one_type]
-                                                   [particle_two_type],
-    normal_force.norm(),
-    normal_unit_vector);
+  if (rolling_resistance_method == Parameters::Lagrangian::ModelParameters::
+                                     RollingResistanceMethod::no_resistance)
```

Comment:
I don't think that's a good idea, because you still have one if per particle. It would be better to templatize the force model itself instead of having the ifs in the force model like that. Here you loose all advantage of templating...